sonic_installer/bootloader/bootloader.py

```diff
@@ -43,8 +52,23 @@ def verify_binary_image(self, image_path):
         """verify that the image is supported by the bootloader"""
         raise NotImplementedError
 
+    def verify_reboot(self):
+        """verify the image for reboot"""
+        image = self.get_next_image()
```

Comment:
@jleveque pointed out on my review that this class should not implement function bodies.
`OnieInstallerBootloader` might be a better candidate.

Reply:
@jleveque Could you share your thought? It looks good to me because it is general/safe/default implementation for SONiC on any platform, not only ONIE but also Aboot.

It is like C++ virtual function vs pure virtual function

---
In reply to: [452473189](https://github.com/Azure/sonic-utilities/pull/979#discussion_r452473189) [](ancestors = 452473189)

Reply:
The idea was to keep the Bootloader class as an abstract base class; all implementation will be in the subclasses.

Reply:
@jleveque Do you think it is ok to implement the checking image path existing in all the subclasses?

Reply:
I think it keeps in line with the current design, and allows different bootloaders to implement different logic.

Reply:
Thanks @jleveque sharing the though flow!
I vote for leave it here for reasons:
1. all bootloaders will share this code
2. Aboot or any other subclass may override it, but still call this base class implementation first, then implement more verification logic.

---
In reply to: [452504725](https://github.com/Azure/sonic-utilities/pull/979#discussion_r452504725) [](ancestors = 452504725,452473189)